test/distributed/_tensor/test_dtensor_compile.py

```diff
@@ -182,20 +186,40 @@ def world_size(self):
         return 4
 
     @with_comms
-    def test_tp_compile_fullgraph(self):
+    @parametrize("is_seq_parallel", [True, False])
+    def test_tp_compile_fullgraph(self, is_seq_parallel):
         mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
 
         model = SimpleModel(self.device_type)
+        module_prepare_input = (
+            PrepareModuleInput()
+            if is_seq_parallel
+            else PrepareModuleInput(input_layouts=Replicate())
+        )
+        no_input_prepare_colwise_style = ColwiseParallel(input_layouts=None)
```

Comment:
hmmm I don't think our `input_layouts` accept None as it's a `Union[Placement, Tuple[Placement]]`?

We should probably just do a check inside prepare_input_fn to assure input_layouts matches the passed in DTensor, can be done in follow up PRs

Reply:
sure, let me first merge this and do in a follow-up PR.